### PR TITLE
chore(evals): add gpt 5.4 mini and nano

### PR DESCRIPTION
**Pull request by ceolinwill**
## Summary by cubic
Add `openai-gpt-5.4` models (base, mini, nano) to the eval suite and refresh model outputs across activities. Also align tests with the visuals schema and handle null discovery fields in grammar steps.

- **New Features**
  - Added eval outputs for `openai-gpt-5.4`, `openai-gpt-5.4-mini`, and `openai-gpt-5.4-nano` across grammar, sentences, vocabulary, sentence-word-translation, pronunciation, and course categories/description; added a Portuguese greetings/farewells vocabulary battle to validate alternative translations.
  - Refreshed model baselines: added `anthropic-claude-4.6` and `google-gemini-3.1-*` previews; removed older entries (e.g., `openai-gpt-5.2`, `openai-gpt-5-mini`, several 4.5/4.1 outputs).

- **Refactors**
  - Tests now expect code visuals to include `annotations: null`.
  - Grammar content step accepts nullable discovery `question`/`context` and normalizes empty values.

<sup>Written for commit e1102fc824561abfe197c2198e2b2244bbf52bd7. Summary will update on new commits.</sup>

### CODE REVIEW

File: packages/ai/src/tasks/steps/_tools/table.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { z } from "zod";
 
 export const tableInputSchema = z.object({
-  caption: z.string().optional().describe("Optional caption (max 100 chars)"),
+  caption: z.string().nullable().describe("Optional caption (max 100 chars)"),
```
[the remark this answers is not in the document]
Comment (ceolinwill) [reply packages/ai/src/tasks/steps/_tools/table.ts:4]: this is by design. openai api requires are fields, so this must be nullable
